pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -169,6 +171,32 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 		return nil, err
 	}
 
+	// Replace the standard k8s emptyDir volume plugin with a wrapper version
+	// which offers XFS quota functionality, but only if the node config
+	// specifies an empty dir quota to apply to projects:
+	if options.ProjectEmptyDirQuota != nil {
+		glog.V(2).Info("Replacing empty-dir volume plugin with quota wrapper")
+		wrappedEmptyDirPlugin := false
+		for idx, plugin := range cfg.VolumePlugins {
+			// Can't really do type checking or use a constant here as they are not exported:
+			if plugin.Name() == "kubernetes.io/empty-dir" {
```

Comment:
I don't think it would be a new volume plugin, we want basically exactly what emptyDir does just with some added functionality, so long term I'd like to see it embedded within k8s emptyDir plugin or volume plugins in general. Doing so as a new plugin would likely be a bunch of copied code that has to be manually maintained.